operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -74,49 +80,50 @@ func firstCharsToLower(s string) string {
-		{"Values.global.proxy.concurrency", "meshConfig.defaultConfig.concurrency", uint32(0)},
-		{"Values.global.proxy.envoyAccessLogService", "meshConfig.defaultConfig.envoyAccessLogService", nil},
-		{"Values.global.proxy.envoyAccessLogService.enabled", "meshConfig.enableEnvoyAccessLogService", nil},
-		{"Values.global.proxy.envoyMetricsService", "meshConfig.defaultConfig.envoyMetricsService", nil},
-		{"Values.global.proxy.protocolDetectionTimeout", "meshConfig.protocolDetectionTimeout", ""},
-		{"Values.pilot.ingress", "meshConfig.ingressService, meshConfig.ingressControllerMode, and meshConfig.ingressClass", nil},
-		{"Values.global.mtls.enabled", "the PeerAuthentication resource", nil},
-		{"Values.global.mtls.auto", "meshConfig.enableAutoMtls", nil},
-		{"Values.grafana.enabled", "the samples/addons/ deployments", false},
-		{"Values.tracing.enabled", "the samples/addons/ deployments", false},
-		{"Values.kiali.enabled", "the samples/addons/ deployments", false},
-		{"Values.prometheus.enabled", "the samples/addons/ deployments", false},
-		{"Values.global.tracer.lightstep.address", "meshConfig.defaultConfig.tracing.lightstep.address", ""},
-		{"Values.global.tracer.lightstep.accessToken", "meshConfig.defaultConfig.tracing.lightstep.accessToken", ""},
-		{"Values.global.tracer.zipkin.address", "meshConfig.defaultConfig.tracing.zipkin.address", nil},
-		{"Values.global.tracer.stackdriver.debug", "meshConfig.defaultConfig.tracing.stackdriver.debug", false},
-		{"Values.global.tracer.stackdriver.maxNumberOfAttributes", "meshConfig.defaultConfig.tracing.stackdriver.maxNumberOfAttributes", 0},
-		{"Values.global.tracer.stackdriver.maxNumberOfAnnotations", "meshConfig.defaultConfig.tracing.stackdriver.maxNumberOfAnnotations", 0},
-		{"Values.global.tracer.stackdriver.maxNumberOfMessageEvents", "meshConfig.defaultConfig.tracing.stackdriver.maxNumberOfMessageEvents", 0},
-		{"Values.global.tracer.datadog.address", "meshConfig.defaultConfig.tracing.datadog.address", ""},
-		{"Values.global.meshExpansion.enabled", "Gateway and other Istio networking resources, such as in samples/istiod-gateway/", false},
-		{"Values.global.trustDomain", "meshConfig.trustDomain", false},
-		{"Values.gateways.istio-ingressgateway.meshExpansionPorts", "components.ingressGateways[name=istio-ingressgateway].k8s.service.ports", nil},
-		{"AddonComponents.grafana.Enabled", "the samples/addons/ deployments", false},
-		{"AddonComponents.tracing.Enabled", "the samples/addons/ deployments", false},
-		{"AddonComponents.kiali.Enabled", "the samples/addons/ deployments", false},
-		{"AddonComponents.prometheus.Enabled", "the samples/addons/ deployments", false},
-		{"AddonComponents.istiocoredns.Enabled", "the in-proxy DNS capturing (ISTIO_META_DNS_CAPTURE)", false},
-		{"Values.istiocoredns.enabled", "the in-proxy DNS capturing (ISTIO_META_DNS_CAPTURE)", false},
+		{"Values.global.certificates", "meshConfig.certificates", nil, false},
```

Comment:
it's almost always false so to express this more compactly, create a map called failHard and put just the two paths there.

Reply:
we may have other fields that would need to fail hard as well, moving to a separate map would introduce some duplicate logic, would prefer to keep it here so can easily define fail hard or not?

Reply:
i think my comment may be unclear... i'm saying you can just create a map like this:
failHard = map[string]bool {
"Values.istiocoredns.enabled": true,
...
}

then 

if failHard[path]...
Otherwise it's really hard to read a bunch of false/trues without field names. Especially so since only two have true value.